multiversx_sdk/controllers/smart_contract_controller.py

```diff
@@ -21,35 +22,8 @@
 from multiversx_sdk.network_providers.transactions import TransactionOnNetwork
 
 
-class IQuery(Protocol):
-    def get_contract(self) -> IAddress:
-        ...
-
-    def get_function(self) -> str:
-        ...
-
-    def get_encoded_arguments(self) -> Sequence[str]:
-        ...
-
-    def get_caller(self) -> Optional[IAddress]:
-        ...
-
-    def get_value(self) -> int:
-        ...
-
-
-class IQueryResponse(Protocol):
-    return_data: List[str]
-    return_code: str
-    return_message: str
-    gas_used: int
-
-    def get_return_data_parts(self) -> List[bytes]:
-        ...
-
-
 class INetworkProvider(Protocol):
-    def query_contract(self, query: IQuery) -> IQueryResponse:
+    def query_contract(self, query: SmartContractQuery) -> SmartContractQueryResponse:
```

Comment:
:+1: Indeed, it's reasonable & practical to have DTOs as input & output of functions in interfaces.